frontend/components/Filter/StudentFilter/StudentFilter.tsx

```diff
@@ -0,0 +1,552 @@
+import styles from "../Filter.module.css";
+import React, { SyntheticEvent, useContext, useEffect, useState } from "react";
+import {
+    EmailStatus,
+    getNextSort,
+    Role,
+    Sort,
+    Student,
+    StudentStatus,
+} from "../../../types";
+import SessionContext from "../../../contexts/sessionProvider";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import CheckIconColor from "../../../public/images/green_check_mark_color.png";
+import CheckIcon from "../../../public/images/green_check_mark.png";
+import ExclamationIconColor from "../../../public/images/exclamation_mark_color.png";
+import ExclamationIcon from "../../../public/images/exclamation_mark.png";
+import ForbiddenIconColor from "../../../public/images/forbidden_icon_color.png";
+import ForbiddenIcon from "../../../public/images/forbidden_icon.png";
+
+export const StudentFilter: React.FC<{
+    setFilteredStudents: (user: Array<Student>) => void;
+}> = ({ setFilteredStudents }) => {
+    const { getSessionKey } = useContext(SessionContext);
+    const router = useRouter();
+
+    const [firstNameFilter, setFirstNameFilter] = useState<string>("");
+    const [lastNameFilter, setLastNameFilter] = useState<string>("");
+    const [emailFilter, setEmailFilter] = useState<string>("");
```

Comment:
@HuanYu-Tan I think we created an enum for this, (last file from this pr), We could use the `EMPTY` value as initial state here

Reply:
These are the values in the textfields. So seems fine as is.